### DISCUSSION TITLE
Problem adding STM32F303 target

### DISCUSSION THREAD

**Discussion by JojoS62** [mbed-ce/mbed-os#126] [General]
I'm trying to add another custom target, the RobotDyn F303CC board, similar to Blackpill F411.
As a starting point, I have added it as custom target like in Jamies repo. The simple project compiles and links, but does not run. I have tried to debug, but the debugger disconnects after a few seconds. I see that it runs from Reset_Handler to SystemInit, and then fails calling  _start.  Looks like it fails in mbed_sdk_boot.c
Any ideas what could be wrong? 

https://github.com/JojoS62/mbed-ce-lvgl

**Comment by JojoS62** [comment]
The RAM size in cmsis_nvic.h is wrong, it is declared as 48k, but real size is 40k. The MCU has a total of 48k, but the ram ranges are not continous.

![grafik](https://user-images.githubusercontent.com/6775562/213722658-1b472619-a190-4600-9fba-6819bf359d92.png)

Will test it later.

**Comment by JojoS62** [comment]
The program starts when I fix the RAM size. But now it fails with hardfault also when sleep_for() is called. Same problem as for F411?

**Reply by JohnK1987** [reply]
Nope, issues what I faced with F411 was caused to my low knowledge with that part of job. Usually a combination of wrong settings from CubeMX and settings in targets.json (custom_targets.json in your case ofc).

BR, Jan

**Reply by JojoS62** [reply]
This problem is also fixed, my mbed_app.json was misspelled with '-' instead of '_'.
Bad that this causes a hardfault.

**Reply by JohnK1987** [reply]
I usually start with HSI, remove LPTICKER and lse_available set to 0, then I continue.

Btw you forgot to make a clock settings for USB
```
#if (DEVICE_USBDEVICE)
  PeriphClkInit.PeriphClockSelection = RCC_PERIPHCLK_USB;
  PeriphClkInit.USBClockSelection = RCC_USBCLKSOURCE_PLL_DIV1_5; // when cpu run 72MHz with HSE 8MHz
  if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInit) != HAL_OK)
  {
    rreturn 0; // FAIL
  }
#endif
```
BR, Jan

**Reply by JojoS62** [reply]
thanks, have not checked everything yet.
